subprojects/functional/src/main/java/org/gradle/internal/lazy/Lazy.java

```diff
@@ -67,6 +67,21 @@ static Factory unsafe() {
         return UnsafeLazy::new;
     }
 
+    /**
+     * A non-atomic {@link Lazy} allows concurrent access from multiple threads without locking.
+     * <br />
+     * The given {@code Supplier} might be executed more than once when multiple threads access a
+     * non-initialized {@link Lazy} at the same time but one value will eventually be cached and the
+     * {@code Supplier} discarded.
+     * <br />
+     * <b>WARNING:</b> Given the above, this flavor of {@code Lazy} initialization should not be used
+     * with a {@code Supplier} that can return different values in scenarios that require a single
+     * stable identity to be observed from all threads.
+     */
+    static Factory nonAtomic() {
```

Comment:
🧐 It might be better to migrate this explanatory comment to the `NonAtomicLazy` class directly, and just `{@link NonAtomicLazy}` here.  In case that class is constructed directly without using the `Lazy` factory.

Reply:
I don't feel strongly either way, as long as the documentation is _there_ and discoverable

Reply:
I kept the comment in `Lazy#atomic()` since it's expected to be the most used contact point with the API. I added a `@see Lazy#atomic()` to the class for quick access.